src/content/post/2024-04-30-kuzu-v-0.4.0.md

```diff
@@ -0,0 +1,262 @@
+```
+
+At any time you can show the databases (and their types) that are attached to a given Kùzu session.
+
+```cypher
+CALL show_attached_databases() RETURN *;
+// Output
+-----------------------------------
+| name            | database type |
+-----------------------------------
+| db1             | POSTGRES      |
+-----------------------------------
+| another_db      | DUCKDB        |
+-----------------------------------
+```
+
+More such extensions are planned in the future to make the process of analyzing graph-like data
+obtained from various sources even more seamless.
+
+### Import/Export database
+
+You can now migrate databases between different Kùzu versions without manually writing DDL and `COPY` statements.
+In the background, Kùzu generates the required DDL statements to facilitate the transfer of data out
+of one version and into another. The `EXPORT DATABASE` command allows you to export the contents of
+the database to a specific directory. The query below exports the database to an absolute path,
+`/path/to/export`, utilizing the same configuration parameters as `COPY FROM` statements.
+
+```cypher
+EXPORT DATABASE TO '/path/to/export' (HEADER=true);
+```
+
+Exporting a database generates three Cypher files, one each for the schema, macro definitions and
+the `COPY` statements for each table. In addition, the data files are exported to CSV (by default),
+or Parquet, if desired.
+
+The `IMPORT DATABASE` command allows you to import the contents of
+a database from a specific directory. The query below imports the database from `/path/to/export`.
+
+```cypher
+IMPORT DATABASE FROM '/path/to/export';
+```
+
+The aim of this functionality is to make it easier to migrate databases between different
+Kùzu versions as we continue to evolve the core and the storage layer in future versions, and to
+facilitate the sharing of databases between users over longer time periods.
+
+### `COPY FROM` with subquery
+
+In v0.4.0, we added support for using subqueries following the `COPY FROM` statement. This feature allows you to
+first perform a task like `MATCH` and then use the results of that query as input to the `COPY FROM` command.
+
+For example, consider that we have a graph with a `User` node label and a `Follows` relationship type.
+We want to create a new `Person` node table and a `Knows` relationship table, where the goal is to state that
+a `Person` "knows" another `Person` if they follow each other. We can use the `COPY FROM` command with a subquery
+to achieve this as follows:
+
+```cypher
+// Create node/rel tables
+CREATE NODE TABLE Person(name STRING, PRIMARY KEY (name));
+CREATE REL TABLE Knows(FROM Person TO Person);
+// Run COPY FROM with a subquery
+COPY Person FROM (MATCH (a:User) RETURN a.name);
+COPY Knows FROM (MATCH (a:User)-[r:Follows]->(b:User) RETURN a.name, b.name);
+```
+
+An alternate use case would be when you want to directly scan data from an existing object, such as
```

Comment:
`directly scan data` -> `directly ingest data`. Later one sounds more accurate since we are in the context of `COPY FROM`